### PR TITLE
Fix qwen3 omni audio lengths #26630

### PR DESCRIPTION
**Pull request by baonudesifeizhai**
FIX #26630
Fix audio processing bug in Qwen3-Omni model that caused split_with_sizes RuntimeError when processing audio inputs. The issue was caused by incorrect calculation of audio_feature_lengths in the model's audio processing pipeline.
 
 

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

File: vllm/model_executor/models/qwen3_omni_moe_thinker.py [hunk 1]
```diff
@@ -733,25 +733,6 @@ def pad_to_hop_length(x: np.ndarray, hop_length: int) -> np.ndarray:
...
-                num_frame = (
-                    (audio_length // hop_length)
-                    if audio_length % hop_length == 0
-                    else (audio_length // hop_length - 1)
-                )
-                audio_num_frames.append(num_frame)
-            hf_inputs["feature_attention_mask"] = [
-                torch.ones(num_frame) for num_frame in audio_num_frames
-            ]
-            hf_inputs["audio_feature_lengths"] = torch.tensor(audio_num_frames)
```
Comment (Isotr0py) [inline vllm/model_executor/models/qwen3_omni_moe_thinker.py:754]: Hmmm, this is intentional to make sure mm_cache can hit audio item correctly, because `padding=True` will create different attention mask when audios are padded/unpadded.

You can simply run the following test for checking:
```
pytest -s -v tests/models/multimodal/processing/test_common.py -k qwen3-omni -v
```
```
  Full diff:
    {
        'audio_feature_lengths': tensor([5, 4]),
  -     'feature_attention_mask': [
  +     'feature_attention_mask': tensor([[1, 1, 1, 1, 1],
  -         tensor([1, 1, 1, 1, 1], dtype=torch.int32),
  ?         -------             ^
  +         [1, 1, 1, 1, 0]], dtype=torch.int32),
  ?                      ^^
  -         tensor([1, 1, 1, 1], dtype=torch.int32),
  -     ],
```

Comment (baonudesifeizhai) [reply vllm/model_executor/models/qwen3_omni_moe_thinker.py:754]: my bad 
if they chunk the audio less then 30s  , it works ..
